lightning/src/ln/blinded_payment_tests.rs

```diff
@@ -1958,3 +1961,206 @@ fn test_trampoline_inbound_payment_decoding() {
+
+	let carol_alice_trampoline_session_priv = secret_from_hex("a0f4b8d7b6c2d0ffdfaf718f76e9decaef4d9fb38a8c4addb95c4007cc3eee03");
+	let carol_blinding_point = PublicKey::from_secret_key(&secp_ctx, &carol_alice_trampoline_session_priv);
+	let carol_blinded_hops = if success {
+		let payee_tlvs = UnauthenticatedReceiveTlvs {
+			payment_secret,
+			payment_constraints: PaymentConstraints {
+				max_cltv_expiry: u32::max_value(),
+				htlc_minimum_msat: amt_msat,
+			},
+			payment_context: PaymentContext::Bolt12Refund(Bolt12RefundContext {}),
+		};
+
+		let nonce = Nonce([42u8; 16]);
+		let expanded_key = nodes[2].keys_manager.get_inbound_payment_key();
+		let payee_tlvs = payee_tlvs.authenticate(nonce, &expanded_key);
+		let carol_unblinded_tlvs = payee_tlvs.encode();
+
+		let path = [(carol_node_id, WithoutLength(&carol_unblinded_tlvs))];
+		blinded_path::utils::construct_blinded_hops(
+			&secp_ctx, path.into_iter(), &carol_alice_trampoline_session_priv
+		).unwrap()
+	} else {
+		let payee_tlvs = blinded_path::payment::TrampolineForwardTlvs {
+			next_trampoline: alice_node_id,
+			payment_constraints: PaymentConstraints {
+				max_cltv_expiry: u32::max_value(),
+				htlc_minimum_msat: amt_msat,
+			},
+			features: BlindedHopFeatures::empty(),
+			payment_relay: PaymentRelay {
+				cltv_expiry_delta: 0,
+				fee_proportional_millionths: 0,
+				fee_base_msat: 0,
+			},
+			next_blinding_override: None,
+		};
+
+		let carol_unblinded_tlvs = payee_tlvs.encode();
+		let path = [(carol_node_id, WithoutLength(&carol_unblinded_tlvs))];
+		blinded_path::utils::construct_blinded_hops(
+			&secp_ctx, path.into_iter(), &carol_alice_trampoline_session_priv
+		).unwrap()
+	};
+
+
+	let route = Route {
+		paths: vec![Path {
+			hops: vec![
+				// Bob
+				RouteHop {
+					pubkey: bob_node_id,
+					node_features: NodeFeatures::empty(),
+					short_channel_id: alice_bob_scid,
+					channel_features: ChannelFeatures::empty(),
+					fee_msat: 1000,
+					cltv_expiry_delta: 48,
+					maybe_announced_channel: false,
+				},
+
+				// Carol
+				RouteHop {
+					pubkey: carol_node_id,
+					node_features: NodeFeatures::empty(),
+					short_channel_id: bob_carol_scid,
+					channel_features: ChannelFeatures::empty(),
+					fee_msat: 0,
+					cltv_expiry_delta: 48,
+					maybe_announced_channel: false,
+				}
+			],
+			blinded_tail: Some(BlindedTail {
+				trampoline_hops: vec![
+					// Carol
+					TrampolineHop {
+						pubkey: carol_node_id,
+						node_features: Features::empty(),
+						fee_msat: amt_msat,
+						cltv_expiry_delta: 24,
+					},
+				],
+				hops: carol_blinded_hops,
+				blinding_point: carol_blinding_point,
+				excess_final_cltv_expiry_delta: 39,
+				final_value_msat: amt_msat,
+			})
+		}],
+		route_params: None,
+	};
+
+	nodes[0].node.send_payment_with_route(route.clone(), payment_hash, RecipientOnionFields::spontaneous_empty(), PaymentId(payment_hash.0)).unwrap();
+
+	if success {
+		check_added_monitors!(&nodes[0], 1);
```

Comment:
this `check_added_monitors` and the one in the else branch can be moved above the if?